wander-embedded-sdk/src/WanderEmbedded.ts

```diff
@@ -0,0 +1,97 @@
+import { setupWalletSDK } from "wallet-api/wallet-sdk.es.js";
+import { WanderEmbeddedOptions } from "./types";
+import { WanderButton } from "./components/WanderButton";
+import { WanderIframe } from "./components/WanderIframe";
+import {
+  IncomingMessage,
+  IncomingMessageId,
+  IncomingResizeMessageData
+} from "./types/messages";
+import { UIComponents } from "./types/embedded";
+
+export class WanderEmbedded {
+  private components: UIComponents;
+  private readonly DEFAULT_IFRAME_SRC = "http://localhost:5174/";
+  private options: WanderEmbeddedOptions;
+
+  constructor(options?: WanderEmbeddedOptions) {
+    this.options = options || {};
+    this.components = this.initializeComponents(options);
+  }
+
+  private initializeComponents(options?: WanderEmbeddedOptions): UIComponents {
+    const iframe = new WanderIframe({
```

Comment:
We should avoid creating this if `options.iframeRef` is provided:

```
if (!options?.iframeRef) {
  const iframe = new WanderIframe({ ... });
  document.body.appendChild(iframe.getElement());
}
```

Messages should be handled in this root class and decoupled from the component classes, which should only care about styling but should be controlled from outside.